addons/binding/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/handler/HyperionHandler.java

```diff
@@ -0,0 +1,315 @@
+            address = (String) config.get(PROP_HOST);
+            port = ((BigDecimal) config.get(PROP_PORT)).intValue();
+            refreshInterval = ((BigDecimal) config.get(PROP_POLL_FREQUENCY)).intValue();
+            priority = ((BigDecimal) config.get(PROP_PRIORITY)).intValue();
+
+            connection = new JsonTcpConnection(address, port);
+            connectFuture = scheduler.scheduleWithFixedDelay(connectionJob, 0, refreshInterval, TimeUnit.SECONDS);
+            refreshFuture = scheduler.scheduleWithFixedDelay(refreshJob, 0, refreshInterval, TimeUnit.SECONDS);
+        } catch (UnknownHostException e) {
+            logger.warn("Could not resolve host: {}", e.getMessage());
+            updateOnlineStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing of Hyperion thing handler.");
+        if (refreshFuture != null) {
+            refreshFuture.cancel(true);
+        }
+        if (connectFuture != null) {
+            connectFuture.cancel(true);
+        }
+        if (connection != null && connection.isConnected()) {
+            try {
+                connection.close();
+            } catch (IOException e) {
+                // do nothing
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        try {
+            if (command instanceof RefreshType) {
+                if (refreshFuture.isDone()) {
+                    refreshFuture = scheduler.scheduleWithFixedDelay(refreshJob, 0, refreshInterval, TimeUnit.SECONDS);
+                } else {
+                    logger.debug("Previous refresh not yet completed");
+                }
+            } else if (CHANNEL_BRIGHTNESS.equals(channelUID.getId())) {
+                handleBrightness(command);
+            } else if (CHANNEL_COLOR.equals(channelUID.getId())) {
+                handleColor(command);
+            } else if (CHANNEL_CLEAR.equals(channelUID.getId())) {
+                handleClear(command);
+            } else if (CHANNEL_EFFECT.equals(channelUID.getId())) {
+                handleEffect(command);
+            }
+        } catch (IOException e) {
+            logger.warn("Unable to send command: {}", command.toString());
+        } catch (CommandUnsuccessfulException e) {
+            logger.warn("Server rejected the command: {}", e.getMessage());
+        }
+    }
+
+    private void handleEffect(Command command) throws IOException, CommandUnsuccessfulException {
+        if (command instanceof StringType) {
+            String effectName = command.toString();
+
+            Effect effect = new Effect(effectName);
+            EffectCommand effectCommand = new EffectCommand(effect, priority);
+            sendCommand(effectCommand);
+        } else {
+            logger.warn("Channel {} unable to process command {}", CHANNEL_EFFECT, command.toString());
```

Comment:
reduce all of those to debug - commands that are not known should simply be ignored by a channel. Maybe there are other channels linked to the same item, which react on the command.

Reply:
done